synapse/storage/databases/main/room.py

```diff
@@ -2132,21 +2135,14 @@ async def upsert_room_on_join(
             # invalid, and it would fail auth checks anyway.
             raise StoreError(400, "No create event in state")
 
-        room_creator = create_event.content.get(EventContentFields.ROOM_CREATOR)
-
-        if not isinstance(room_creator, str):
-            # If the create event does not have a creator then the room is
-            # invalid, and it would fail auth checks anyway.
-            raise StoreError(400, "No creator defined on the create event")
-
         await self.db_pool.simple_upsert(
             desc="upsert_room_on_join",
             table="rooms",
             keyvalues={"room_id": room_id},
             values={"room_version": room_version.identifier},
             insertion_values={
                 "is_public": False,
-                "creator": room_creator,
```

Comment:
Should this key on room version?

Reply:
To obey the letter of the law, yes. Though as discussed, I think Synapse has been effectively disallowing creator != sender forever so I'm not too worried about this.